docs/howtos/working_with_urls_and_api_endpoints.md

```diff
@@ -0,0 +1,423 @@
+# Working with URLs and API Endpoints
+
+This guide explains how to work with URLs and API endpoints in Kolibri, including the URL namespacing system used throughout the codebase.
+
+## Overview
+
+Kolibri uses a consistent URL namespacing pattern that bridges Django's backend URL system with JavaScript frontend code. This allows type-safe URL construction with automatic parameter validation across the entire stack.
+
+## URL Naming Convention
+
+All URLs in Kolibri follow the namespace pattern:
+
+```
+kolibri:<app_name>:<resource_name>_<endpoint>
+```
+
+### Core URLs
+
+Core Kolibri URLs use the `core` namespace:
```

Comment:
If we're going to refer to this as `namespace` why do we call it `app_name` above? We should be consistent here.